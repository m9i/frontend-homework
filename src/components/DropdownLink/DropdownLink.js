import React from "react";
import {connect} from "react-redux";
import {toggleDropdownAction} from "../../store/actions/toggleDropdownAction";

const DropdownLink = ({toggle, showDropdown, hideDropdown, selectedCompany}) => {

    const toggleDropdown = () => {
        toggle ? hideDropdown() : showDropdown()
    }

    return (
        <div className="nav__link" onClick={() => toggleDropdown()}>
            <div className="nav__link-text-wrapper">
                <div className="nav__link-text">Elon Musk</div>
                <div className="nav__link-subtext">
                    {selectedCompany.value}
                </div>
            </div>
            <i className="material-icons-outlined nav__link-icon">settings</i>
        </div>
    );
};

const mapStateToProps = (state) => ({
    selectedCompany: state.company.selectedCompany,
    toggle: state.dropdown.showDropdown
});

const mapDispatchToProps = (dispatch) => ({
    onShowDropdown: () => dispatch(toggleDropdownAction()),
    hideDropdown: () => dispatch(toggleDropdownAction(false)),
    showDropdown: () => dispatch(toggleDropdownAction(true))
});

export default connect(mapStateToProps, mapDispatchToProps)(DropdownLink);
