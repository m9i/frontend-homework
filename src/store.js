import { createStore, applyMiddleware } from "redux";
import thunkMiddleware from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import reducer from "./reducers/reducer";

const composedEnhancer = composeWithDevTools(applyMiddleware(thunkMiddleware));
const store = createStore(reducer, composedEnhancer);

store.subscribe(() => {
  console.log("subscribe", store.getState());
});
export default store;
