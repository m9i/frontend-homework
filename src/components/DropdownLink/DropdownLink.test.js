import React from "react"
import DropdownLink from "./DropdownLink"
import { shallow, mount } from "enzyme"
import store from "../../store"
import { Provider } from "react-redux"

describe("DropdownLink", () => {
  it("snapshot", () => {
    const DropdownLinkTree = shallow(
      <Provider store={store}>
        <DropdownLink />
      </Provider>
    ).dive()
    expect(DropdownLinkTree).toMatchSnapshot()
  })
})
