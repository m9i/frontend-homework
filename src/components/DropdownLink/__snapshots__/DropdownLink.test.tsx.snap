// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownLink snapshot 1`] = `<Connect(DropdownLink) />`;
